src/classes/UTIL_Describe.cls

```diff
@@ -660,6 +720,108 @@ public class UTIL_Describe {
+     * @return Schema.RecordTypeInfo The RecordTypeInfo for a given recordTypedevName, object
+     */
+    public static Schema.RecordTypeInfo getRecordTypeInfoByDeveloperName(String objectName, String recordTypeDevName) {
+        if (recordTypeInfoByDeveloperName.get(recordTypedevName) == null) {
+     
+            // Get the Describe Result
+            fillMapsForObject(objectName);
+            Schema.DescribeSObjectResult obj = objectDescribes.get(objectName);
+
+            // Obtain ALL Active Record Types for the given SObjectType objectType
+            String soql = 'SELECT Id, Name, DeveloperName FROM RecordType ' + 'WHERE SObjectType = \''
+                          + String.escapeSingleQuotes(obj.getName()) + '\' ' + 'AND IsActive = TRUE';
+            List<SObject> results;
+
+            try {
+                results = Database.query(soql);
+            } catch (Exception ex) {
+                results = new List<SObject>();
+            }
+
+            // Obtain the RecordTypeInfos for this SObjectType objectType
+            Map<Id, Schema.RecordTypeInfo> recordTypeInfos = obj.getRecordTypeInfosByID();
+
+            for (SObject rt : results) {
+
+                // Check RecordType IS available for the running user
+                if (!Test.isRunningTest()) {
+                    if (recordTypeInfos.get(rt.Id).isAvailable()) {
+                        recordTypeInfoByDeveloperName.put(String.valueOf(rt.get('DeveloperName')), recordTypeInfos.get(rt.Id));
+                    }
+                // We don't want our tests to depend on the profile configuration in the org. Otherwise tests
+                // might fail just because the user running them does not have certain record types assigned.
+                } else {
+                    recordTypeInfoByDeveloperName.put(String.valueOf(rt.get('DeveloperName')), recordTypeInfos.get(rt.Id));
+                }
+            }
+        }
+
+        return recordTypeInfoByDeveloperName.get(recordTypeDevName);
+    }
+    
+    /*******************************************************************************************************************
+    * @description Cache for the map recordTypeInfoByName
+    */
+    @testVisible
+    private static Map<String, Schema.RecordTypeInfo> recordTypeInfoByName = new Map<String, Schema.RecordTypeInfo>();
+
+    /**********************************************************************************************************************
+     * @description Returns Schema.RecordTypeInfo for active, user-available recordTypeInfos for a given recordTypeName
+     * @param objectName Name of the object to retrieve recordTypeInfos for
+     * @param recordTypedevName The record type's developer name
+     * @return Schema.RecordTypeInfo The RecordTypeInfo for a given recordTypedevName, object
+     */
+    public static Schema.RecordTypeInfo getRecordTypeInfoByName(String objectName, String recordTypeName) {
```

Comment:
Please collapse this to use the helper method described above.

The end state should look similar to the following.
```
        if (recordTypeInfoByName.get(objectName) == null) {
            populateRecordTypeInfoForSObject(objectName);
        }
        return recordTypeInfoByName.get(objectName).get(recordTypeName);
```